apps/services/party-letter-registry-api/src/app/modules/partyLetterRegistry/partyLetterRegistry.controller.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  Body,
+  Controller,
+  Get,
+  NotFoundException,
+  Post,
+  Query,
+} from '@nestjs/common'
+import { CreateDto } from './dto/create.dto'
+import { FindByOwnerDto } from './dto/findByOwner.dto'
+import { PartyLetterRegistry } from './partyLetterRegistry.model'
+import { PartyLetterRegistryService } from './partyLetterRegistry.service'
+
+@Controller('party-letter-registry')
+export class PartyLetterRegistryController {
+  constructor(
+    private readonly partyLetterRegistryService: PartyLetterRegistryService,
+  ) {}
+
+  @Get()
+  async findByOwner(
+    @Query() { owner }: FindByOwnerDto,
+  ): Promise<PartyLetterRegistry> {
+    const resource = await this.partyLetterRegistryService.findByOwner(owner)
+
+    if (!resource) {
+      throw new NotFoundException("This resource doesn't exist")
+    }
+
+    return resource
+  }
+
+  @Post()
```

Comment:
These endpoints should have OAuth2 authentication with some scopes.

If the client is another service like the application system (and not a user with these scopes), then it would need client credentials to get an access token with these scopes.

Reply:
I don't think the current timeframe allows for implementation of OAuth2, but I see where you are coming from.
Our goal is to present a working system by thursday.
After that deadline the user flows and functionality need to be tested probably accompanied by some iterations.
I'm confident we can implement auth into this service during that timeframe